contribs/gmf/src/directives/editfeatureselector.js

```diff
@@ -147,29 +147,67 @@ gmf.EditfeatureselectorController = function($scope, gmfThemes,
    */
   this.selectedLayer = null;
 
+  $scope.$watch(
+    function() {
+      return this.selectedLayer;
+    }.bind(this),
+    function(newValue, oldValue) {
+      if (newValue) {
+        this.selectedWMSLayer = newValue ? this.wmsLayers_[newValue.id] : null;
+      } else {
+        this.selectedWMSLayer = null;
+        this.state = gmf.EditfeatureController.State.IDLE;
+      }
+    }.bind(this)
+  );
+
   /**
    * The currently selected OpenLayers layer object.
    * @type {?ol.layer.Image|ol.layer.Tile}
    * @export
    */
   this.selectedWMSLayer = null;
 
-  this.themesChangeListenerKey = ol.events.listen(this.gmfThemes_,
-      gmf.ThemesEventType.CHANGE, this.setLayersFromThemes_, this);
-
-  this.registerLayer_(this.dataLayerGroup_);
-
-  $scope.$on('$destroy', this.handleDestroy_.bind(this));
+  /**
+   * The state of this directive shared with the `gmf-editfeature` directive.
+   * This property allows the proper management of the "stop editing" button.
+   * When clicked, the according state is set and the `gmf-editfeature`
+   * directive checks if it has unsaved changes and allow this directive to
+   * continue the action that was made or not.
+   * @type {string}
+   * @export
+   */
+  this.state = gmf.EditfeatureController.State.IDLE;
```

Comment:
No, it i shared.  See: https://github.com/camptocamp/ngeo/pull/1757/files/14cdfe70e12470ef688562c7ad33b6162801859a#diff-4ba85fa994d5b0d6688587a53d2c140dR32